src/translator/response.h

```diff
@@ -40,58 +46,57 @@ class Response {
+      : source(std::move(other.source)), target(std::move(other.target)),
+        histories_(std::move(other.histories_)){};
 
   // Prevents CopyConstruction and CopyAssignment. sourceRanges_ is constituted
   // by string_view and copying invalidates the data member.
   Response(const Response &) = delete;
   Response &operator=(const Response &) = delete;
 
-  typedef std::vector<std::pair<const string_view, const string_view>>
-      SentenceMappings;
+  const size_t size() const { return source.numSentences(); }
+  const Histories &histories() const { return histories_; }
 
-  // Moves source sentence into source, translated text into translation.
-  // Pairs of string_views to corresponding sentences in
-  // source and translation are loaded into sentenceMappings. These string_views
-  // reference the new source and translation.
-  //
-  // Calling move() invalidates the Response object as ownership is transferred.
-  // Exists for moving strc
-  void move(std::string &source, std::string &translation,
-            SentenceMappings &sentenceMappings);
+  std::vector<float> wordQuality(size_t sentenceIdx) {
+    auto wordQualities = hypothesis(sentenceIdx)->tracebackWordScores();
+    // Remove the eos token.
+    wordQualities.pop_back();
+    return wordQualities;
+  }
 
-  const Histories &histories() const { return histories_; }
-  const std::string &source() const { return source_; }
-  const std::string &translation() {
-    constructTranslation();
-    return translation_;
+  float sentenceQuality(size_t sentenceIdx) {
+    NBestList onebest = histories_[sentenceIdx]->nBest(1);
+    Result result = onebest[0]; // Expecting only one result;
+    auto normalizedPathScore = std::get<2>(result);
+    return normalizedPathScore;
   }
 
-  // A convenience function provided to return translated text placed within
-  // source's structure. This is useful when the source text is a multi-line
-  // paragraph or string_views extracted from structured text like HTML and it's
-  // desirable to place the individual sentences in the locations of the source
-  // sentences.
-  // const std::string translationInSourceStructure();
-  // const PendingAlignmentType alignment(size_t idx);
+  const SoftAlignment softAlignment(size_t sentenceIdx) {
+    return hypothesis(sentenceIdx)->tracebackAlignment();
+  }
 
-private:
-  void constructTranslation();
-  void constructSentenceMappings(SentenceMappings &);
+  const WordAlignment hardAlignment(size_t sentenceIdx, float threshold = 1.f) {
+    return data::ConvertSoftAlignToHardAlign(softAlignment(sentenceIdx),
+                                             threshold);
+  }
 
-  std::string source_;
-  SentenceRanges sourceRanges_;
+  AnnotatedBlob source;
+  AnnotatedBlob target;
+
+  const std::string &translation() {
+    LOG(info, "translation() will be deprecated now that target is public.");
+    return target.blob;
+  }
+
+private:
   Histories histories_;
```

Comment:
This needs to die.

Reply:
Will be covered in a different PR. Killing histories means updating marian-decoder replacement and concurrent apps to use the processed stuff.